src/server/network/network.ts

```diff
@@ -0,0 +1,93 @@
+import dgram = require('dgram');
+
+export type ClientListener = (event: string, ...args: any[]) => void;
+
+type TeamData = {
+  server: dgram.Socket;
+  name: string;
+  address: string;
+  port: number;
+};
+
+export class UDPServer {
+  private redTeam: TeamData;
+  private blueTeam: TeamData;
+  private clients: Map<string, ClientListener>;
+
+  constructor(redTeamAddress: string, blueTeamAddress: string) {
+    this.clients = new Map();
+
+    this.redTeam = {
+      server: dgram.createSocket({ type: 'udp4', reuseAddr: true }),
+      name: 'Red Team',
+      address: redTeamAddress,
+      port: 9917,
+    };
+    this.blueTeam = {
+      server: dgram.createSocket({ type: 'udp4', reuseAddr: true }),
+      name: 'Blue Team',
```

Comment:
I'd make this configurable too. Could be useful if people want to setup this server instead to have a multicast address/port per team like "NUbots" rather than just red and blue